config/types/systemd.go

```diff
@@ -56,8 +62,43 @@ func init() {
 				})
 			}
 
+			unitRep := validateUnit(newUnit)
+			rep.Merge(unitRep)
 			out.Systemd.Units = append(out.Systemd.Units, newUnit)
 		}
-		return out, report.Report{}, ast
+		return out, rep, ast
 	})
 }
+
+func validateUnit(unit ignTypes.Unit) report.Report {
+	var rep report.Report
+
+	isEnabled := unit.Enable || (unit.Enabled != nil && *unit.Enabled)
+
+	if unit.Contents != "" {
+		parsedUnit, err := sdunit.Deserialize(strings.NewReader(unit.Contents))
+		if err != nil {
+			rep.Add(report.Entry{
+				Kind:    report.EntryError,
+				Message: fmt.Sprintf("systemd unit %q could not be parsed: %v", unit.Name, err),
+			})
+			return rep
+		}
+
+		noInstall := true
+		for _, section := range parsedUnit {
+			if section.Section == "Install" {
+				noInstall = false
+			}
+		}
+
+		if isEnabled && noInstall {
+			rep.Add(report.Entry{
+				Kind:    report.EntryWarning,
+				Message: fmt.Sprintf("systemd unit %q has no [Install] section; 'enabled' will do nothing", unit.Name),
```

Comment:
Should the warning be `'enable(d)' will do nothing` as this triggers if the config has either?

Reply:
`enable` is deprecated, so I chose to just call it by the supported name